lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -2377,149 +2749,208 @@ int s1ap_mme_handle_enb_reset(
-          calloc(1, sizeof(S1ap_UE_associatedLogicalS1_ConnectionItemRes_t));
+      S1ap_UE_associatedLogicalS1_ConnectionItemResAck_t* sig_conn_item =
+          calloc(1, sizeof(S1ap_UE_associatedLogicalS1_ConnectionItemResAck_t));
+      sig_conn_item->id =
+          S1ap_ProtocolIE_ID_id_UE_associatedLogicalS1_ConnectionItem;
+      sig_conn_item->criticality = S1ap_Criticality_ignore;
+      sig_conn_item->value.present =
+          S1ap_UE_associatedLogicalS1_ConnectionItemResAck__value_PR_UE_associatedLogicalS1_ConnectionItem;
+      S1ap_UE_associatedLogicalS1_ConnectionItem_t* item =
+          &sig_conn_item->value.choice.UE_associatedLogicalS1_ConnectionItem;
       if (enb_reset_ack_p->ue_to_reset_list[i].mme_ue_s1ap_id !=
           INVALID_MME_UE_S1AP_ID) {
-        mme_ue_id  = calloc(1, sizeof(S1ap_MME_UE_S1AP_ID_t*));
-        *mme_ue_id = enb_reset_ack_p->ue_to_reset_list[i].mme_ue_s1ap_id;
-        sig_conn_list->uE_associatedLogicalS1_ConnectionItem.mME_UE_S1AP_ID =
-            mme_ue_id;
+        item->mME_UE_S1AP_ID = calloc(1, sizeof(S1ap_MME_UE_S1AP_ID_t));
+        *item->mME_UE_S1AP_ID =
+            enb_reset_ack_p->ue_to_reset_list[i].mme_ue_s1ap_id;
+      } else {
+        item->mME_UE_S1AP_ID = NULL;
       }
       if (enb_reset_ack_p->ue_to_reset_list[i].enb_ue_s1ap_id !=
           INVALID_ENB_UE_S1AP_ID) {
-        enb_ue_id  = calloc(1, sizeof(S1ap_ENB_UE_S1AP_ID_t));
-        *enb_ue_id = enb_reset_ack_p->ue_to_reset_list[i].enb_ue_s1ap_id;
-        sig_conn_list->uE_associatedLogicalS1_ConnectionItem.eNB_UE_S1AP_ID =
-            enb_ue_id;
+        item->eNB_UE_S1AP_ID = calloc(1, sizeof(S1ap_ENB_UE_S1AP_ID_t));
+        *item->eNB_UE_S1AP_ID =
+            enb_reset_ack_p->ue_to_reset_list[i].enb_ue_s1ap_id;
+      } else {
+        item->eNB_UE_S1AP_ID = NULL;
       }
-      sig_conn_list->uE_associatedLogicalS1_ConnectionItem.iE_Extensions = NULL;
-      ASN_SEQUENCE_ADD(
-          &s1ap_ResetAcknowledgeIEs_p
-               ->uE_associatedLogicalS1_ConnectionListResAck,
-          sig_conn_list);
+      ASN_SEQUENCE_ADD(&ie_p->list, sig_conn_item);
     }
   }
-  if (s1ap_mme_encode_pdu(&message, &buffer, &length) < 0) {
-    OAILOG_ERROR_UE(LOG_S1AP, imsi64, "Reset Ack encoding failed \n");
+  if (s1ap_mme_encode_pdu(&pdu, &buffer, &length) < 0) {
+    OAILOG_ERROR(LOG_S1AP, "Failed to S1 Reset command \n");
+    /** We rely on the handover_notify timeout to remove the UE context. */
+    DevAssert(!buffer);
     OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
   }
-  bstring b = blk2bstr(buffer, length);
-  free_wrapper((void**) &buffer);
-  rc = s1ap_mme_itti_send_sctp_request(
-      &b, enb_reset_ack_p->sctp_assoc_id, enb_reset_ack_p->sctp_stream_id,
-      INVALID_MME_UE_S1AP_ID);
 
-  free_wrapper((void**) &(enb_reset_ack_p->ue_to_reset_list));
-  free_s1ap_resetacknowledge(s1ap_ResetAcknowledgeIEs_p);
-  increment_counter("s1_reset_from_enb", 1, 1, "action", "reset_ack_sent");
```

Comment:
Do not remove the `increment_counter` call as this is used to collect statistics from the MME

Reply:
Resolved